packages/react-core/src/components/NumberInput/examples/WithStatus.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { NumberInput, ValidatedOptions } from '@patternfly/react-core';
+
+export const WithStatus: React.FunctionComponent = () => {
+  const max = 10;
+  const min = 0;
+
+  const [validated, setValidated] = React.useState<ValidatedOptions>(ValidatedOptions.success);
+  const [value, setValue] = React.useReducer((state, newVal) => Math.max(min, Math.min(max, Number(newVal))), 5);
+
+  const onPlus = () => {
+    const newVal = value + 1;
+    setValue(newVal);
+    validate(newVal);
+  };
+
+  const onMinus = () => {
+    const newVal = value - 1;
+    setValue(newVal);
+    validate(newVal);
+  };
+
+  const onChange = event => {
+    const newVal = isNaN(event.target.value) ? 5 : Number(event.target.value);
+    setValue(newVal);
+    validate(newVal);
+  };
+
+  const validate = newVal => {
+    const diff = Math.abs(5 - newVal);
+    if (diff > 3) {
+      setValidated(ValidatedOptions.error);
+    } else if (diff > 1) {
+      setValidated(ValidatedOptions.warning);
+    } else {
+      setValidated(ValidatedOptions.success);
+    }
+  };
+  return (
+    <>
+      Status will change to warning/error depending on how far from 5.
```

Comment:
Nit: this could be reworded to something more along the lines of:

```suggestion
In the following example, the validated status will update based on the value of the number input and how far from the number 5 it is.
```